test/unit/test_utils.py

```diff
@@ -136,3 +136,11 @@ def test_mysql_connection_info_from_uri_password_length(
             MySQLConnectionInfo.from_uri(uri)
     else:
         MySQLConnectionInfo.from_uri(uri)
+
+
+def test_mysql_connection_info_from_uri_unquote_username_and_password() -> None:
+    uri = "mysql://test%40example.com:%40%26%20%7B@<ip>:1234/?ssl-mode=DISABLED"
+    conn_info = MySQLConnectionInfo.from_uri(uri)
```

Comment:
Can we check that `to_uri` also works? Or is there a test covering that?

Reply:
Here is the check:
https://github.com/Aiven-Open/aiven-mysql-migrate/pull/35/files#diff-f8892f46b18e3f8c42e6ca94750dae62f678c1cdfee66955aa713d2a59b27ba6R146

Reply:
Going blind, it seems :crying_cat_face: